[tox]
envlist = py36, flake8, black, pylint

[travis]
python =
  3.6: py36, flake8, black, pylint

[testenv]
deps = pytest
commands =
    python -m pytest

[testenv:flake8]
basepython = python3
skip_install = true
deps = flake8
commands =
    flake8

# TODO: C0121 is a false positive non applicable to Pandas dataframes.
# Solved in flake8 by https://github.com/PyCQA/pycodestyle/issues/450
# Need to open an issue on pylint github
# TODO: cover the tests with pylint too
[testenv:pylint]
deps = pylint
extension-pkg-whitelist = datrie
commands =
    pylint {toxinidir}/auto_complete_server/ --extension-pkg-whitelist=datrie -d C0121

[testenv:black]
basepython = python3
skip_install = true
deps = black
commands =
    black -l 79 --check --diff {toxinidir}/auto_complete_server/
    black -l 79 --check --diff {toxinidir}/tests/
    black -l 79 --check --diff {toxinidir}/setup.py
